### ISSUE TITLE
Add ability to customize Minttl field of SOA record

### ISSUE THREAD

**Issue by tantra35** [hashicorp/consul#2301]
Now this value is hardcoded in consul:

```
// addSOA is used to add an SOA record to a message for the given domain
func (d *DNSServer) addSOA(domain string, msg *dns.Msg) {
    soa := &dns.SOA{
        Hdr: dns.RR_Header{
            Name:   domain,
            Rrtype: dns.TypeSOA,
            Class:  dns.ClassINET,
            Ttl:    0,
        },
        Ns:      "ns." + domain,
        Mbox:    "postmaster." + domain,
        Serial:  uint32(time.Now().Unix()),
        Refresh: 3600,
        Retry:   600,
        Expire:  86400,
        Minttl:  0,
    }
    msg.Ns = append(msg.Ns, soa)
} 
```

But as dns interface of consul not so fast as desired. If service disappear from cluster(for example failure, in aws needs some time while autoscale launch new instance, this takes about 2 minutes) dnmasq doesn't cache negative responces from consul (negttl option doesn't helps, because consul return SOA record) and this leads to huge load on consul

**Comment by tantra35** [comment]
Also consul must set TTL for SOA record itself equal to Minttl. as described in rfc 2308

**Comment by alinoeabrassart** [comment]
I have made the modification in my code adding a Minttl field in DNSConfig in configuration file and reusing it in the addSOA function. If anyone is interested and/or willing to review my code, I could make a PR. Let me know...

**Comment by tantra35** [comment]
@alinoeabrassart could you provide a PR?

**Comment by blake** [comment]
Fixed in #4714.
